step-04/README.md

```diff
@@ -0,0 +1,25 @@
+# step-04
+
+This application builds on step-03 with an example of using structured outputs from LLMs.
+
+## Using the application
+
+1. Access the application locally at http://localhost:8080/
+2. Launch the chatbot by clicking on the agent icon in the bottom right corner of the UI.
+3. Toggle streaming on/off in the chatbot dialog and observe the difference in the output when asking the chatbot questions, such as:
+  - What's a good planet to visit if I want to see volcanoes?
+  - Where should I go if I want to be able to see the earth rise/set?
+  - What's a good place to go to get out of the cold?
+4. Experiment with RAG by asking questions related to the "Terms of Use" document:
+  - Can I get a refund if I cancel my booking 9 days in advance?
+  - I locked my keys in the spaceship
+  - I'm on Neptune and my spaceship broke down
+5. Utlize structured outputs to determine compatible spaceships by asking questions such as:
+  -  Which spaceship can fit 6 passengers and travel to Pluto?
+  -  Which spaceship has room for cargo and can travel to Mars and Saturn?
+  -  Which spaceship can travel outside of our solar system with 3 passangers?
```

Comment:
passangers -> passengers
can it handle more varied questions? (these 3 are quite similar)
eg.  I want to go to Saturn and collect some souvenir rock and ice from the rings. which spaceship is best for that?